multimap/src/main/java/org/infinispan/multimap/impl/EmbeddedMultimapPairCache.java

```diff
@@ -194,4 +226,45 @@ public final CompletionStage<Integer> remove(K key, Collection<HK> properties) {
       if (properties.isEmpty()) return CompletableFuture.completedFuture(0);
       return readWriteMap.eval(key, new HashMapRemoveFunction<>(properties));
    }
+
+   /**
+    * Select {@param count} key-value pairs from the hash map stored under the given key.
+    * <p>
+    * The returned values follow the iteration order of {@link HashMap#entrySet()}. An {@param count} bigger than the
+    * size of the stored map does not raise an exception.
+    *
+    * @param key: Cache key to retrieve the stored hash map.
+    * @param count: Number of key-value pairs to select.
+    * @return {@link CompletionStage} containing a {@link Map} with the key-value pairs or null if not found.
+    */
+   public CompletionStage<Map<HK, HV>> subSelect(K key, int count) {
+      requireNonNull(key, ERR_KEY_CAN_T_BE_NULL);
+      requirePositive(count, ERR_COUNT_MUST_BE_POSITIVE);
+      return cache.getCacheEntryAsync(key)
+            .thenApply(entry -> {
+               if (entry == null) return null;
+
+               return entry.getValue().converted().entrySet().stream()
```

Comment:
if count is equal to or greater can we just return the map?

Reply:
Yep, updated.